### PR TITLE
DAOS-623 test: remove MappedParameter

### PR DESCRIPTION
**Pull request by daltonbohning**
Test-tag: pr,vm datamover,vm
Skip-unit-tests: true
Skip-fault-injection-test: true

Remove MappedParameter class and move functionality to BasicParameter.

Required-githooks: true

Signed-off-by: Dalton Bohning <dalton.bohning@intel.com>

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficent testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by daltonbohning** [comment]
2 existing datamover failures: https://daosio.atlassian.net/browse/DAOS-11909

**Review by shimizukko — APPROVED** [review]

**Review by phender — APPROVED** [review]

**Review by dinghwah — APPROVED** [review]

**Comment by daltonbohning** [comment]
@daos-stack/daos-gatekeeper Please ignore for now. I forgot to run an additional datamover test

File: utils/cq/words.dict [hunk 1]
```diff
@@ -238,6 +239,7 @@ libgurt
 libioil
 libs
 libvirt
+linux
```
Comment (ashleypittman) [inline utils/cq/words.dict:243]: From memory Linux is accepted so it's probably better to change any code from linux to Linux and not make this change.

Comment (daltonbohning) [reply utils/cq/words.dict]: Done. Thanks

**Comment by daltonbohning** [comment]
The relevant datamover tests that use the `mapped_values` functionality passed:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10619/5/testReport/FTEST_datamover/DmvrPosixLargeFile/
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10619/5/testReport/FTEST_datamover/DmvrLargeDir/

As well as all pr and datamover on VM:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-10619/5/testReport/

**Review by shimizukko — APPROVED** [review]

**Review by dinghwah — APPROVED** [review]

**Comment by daltonbohning** [comment]
@daos-stack/daos-gatekeeper Ready now after running appropriate tests :)

**Pull request merged**